jekyll/_cci2/executor-types.md

```diff
@@ -8,23 +8,28 @@ order: 10
 ---
 [building-docker-images]: {{ site.baseurl }}/2.0/building-docker-images/
 
-This document describes the `docker`, `machine`, and `macos` environments in the following sections:
+This document describes the available executor types (`docker`, `machine`, and `macos`) in the following sections:
 
 * TOC
 {:toc}
 
 ## Overview
 {:.no_toc}
 
-CircleCI enables you to run jobs
-in one of three environments:
+An *executor type* defines the underlying technology or environment in which to run a job. CircleCI enables you to run jobs in one of three environments:
 
 - Within Docker images (`docker`)
 - Within a Linux virtual machine (VM) image (`machine`)
 - Within a macOS VM image (`macos`)
 
-For building on Linux,
-there are tradeoffs to using `docker` versus `machine`, as follows:
+It is possible to specify a different executor type for every job in your ['.circleci/config.yml']({{ site.baseurl }}/2.0/configuration-reference/) by specifying the executor type and an appropriate image. An *image* is a packaged system that has the instructions for creating a running environment.  A *container* or *virtual machine* is the term used for a running instance of an image. For example, you could specify an executor type and an image for every job:
+
+- Jobs that require Docker images (`docker`) may use an image for Node.js or Python. The [pre-built CircleCI Docker image]({{ site.baseurl }}/2.0/circleci-images/) from the CircleCI Dockerhub will help you get started quickly without learning all about Docker. These images are not a full operating system, so they will generally make building your software more efficient. 
+- Jobs that require a complete Linux virtual machine (VM) image (`machine`) may use an Ubuntu version such as 16.04.
+- Jobs that require a macOS VM image (`macos`) may use an Xcode version such as 10.0.0.
```

Comment:
I'm curious what this new paragraph adds; I personally found it a bit confusing to parse the combination of specific examples ("may use an image for Node.js or Python") and generally accurate statements ("may use an Ubuntu version _such as_ 16.04").